runtime/parser/parse_model.go

```diff
@@ -237,6 +243,25 @@ func (p *Parser) parseModel(ctx context.Context, node *Node) error {
 	r.ModelSpec.OutputConnector = outputConnector
 	r.ModelSpec.OutputProperties = outputPropsPB
 
+	if r.ModelSpec != nil && len(tmp.Tests) > 0 {
+		var sqlTests []*runtimev1.ModelTest
+		for i := range tmp.Tests {
+			t := &tmp.Tests[i]
+			resolver, props, refs, err := p.parseDataYAML(&t.DataYAML, outputConnector)
+			if err != nil {
+				return fmt.Errorf(`failed to parse test %q: %w`, t.Name, err)
+			}
```

Comment:
You can't return errors here – look for this comment above:
```
	// NOTE: After calling insertResource, an error must not be returned. Any validation should be done before calling it.
```

So move construction of `var sqlTests` up before that, and then just do assignment here